csharp/src/Ice/OutgoingRequestFrame.cs

```diff
@@ -73,25 +76,28 @@ public OutgoingRequestFrame(IObjectPrx proxy, string operation, bool idempotent,
 
             if (context != null)
             {
-                // Explicit context
                 Context = new Context(context);
             }
             else
             {
                 if (proxy.Context != null)
                 {
                     Context = new Context(proxy.Context);
+                    foreach (KeyValuePair<string, string> e in proxy.Communicator.CurrentContext)
+                    {
+                        try
+                        {
+                            Context.Add(e.Key, e.Value);
+                        }
+                        catch (System.ArgumentException)
+                        {
+                            // Ignore duplicate: the proxy Context entry prevails.
+                        }
+                    }
                 }
                 else
                 {
-                    Context = new Context();
-                }
-
-                // TODO: simplify implicit context
-                var implicitContext = (ImplicitContext?)proxy.Communicator.GetImplicitContext();
-                if (implicitContext != null)
-                {
-                    Context = implicitContext.Combine(Context);
+                    Context = new Context(proxy.Communicator.CurrentContext);
                 }
```

Comment:
I think we can simplify this a bit more
```
if (context != null)
{
    Context = new Context(context);
}
else if (proxy.Context != null)
{
    Context = new Context(proxy.Communicator.CurrentContext);
    foreach ((string key, string value) in proxy.Context)
    {
        Context[key] = value;
    }
}
else
{
    Context = new Context(proxy.Communicator.CurrentContext);
}
```

Reply:
Fixing. The `if (proxy.Context != null)` is actually incorrect, as the Context cannot be null.